src/infrastructure/utils/ZodValidatorMiddleware.ts

```
@@ -0,0 +1,224 @@
+import type { Context, MiddlewareHandler, Next } from "hono";
+import { validator } from "hono/validator";
+import type { z } from "zod";
+
+/**
+ * Creates a middleware that validates request parameters using Zod schemas
+ */
+export const zodValidator = <
+  T extends {
+    params?: z.ZodType<any>;
```

Biome (1.9.4)

[error] Unexpected any. Specify a different type.

any disables many type checking rules. Its use should be avoided.

(lint/suspicious/noExplicitAny)